crates/nargo_cli/src/cli/compile_cmd.rs

```diff
@@ -53,17 +53,22 @@ fn save_and_preprocess_program(
     circuit_dir: &Path,
 ) -> Result<(), CliError> {
     save_program_to_file(compiled_program, circuit_name, circuit_dir);
-    preprocess_with_path(circuit_name, circuit_dir, &compiled_program.circuit)?;
+
+    let preprocessed_data = PreprocessedData::from(&compiled_program.circuit);
+    save_preprocess_data(&preprocessed_data, circuit_name, circuit_dir)?;
     Ok(())
 }
 
 /// Save a contract to disk along with proving and verification keys.
 /// - The contract ABI is saved as one file, which contains all of the
 /// functions defined in the contract.
 /// - The proving and verification keys are namespaced since the file
-/// could contain multiple contracts with the same name.
+/// could contain multiple contracts with the same name. The verification key is saved inside
+/// of the ABI.
+// TODO: we also save it to disk, so its saved twice, find a nice abstraction for this
+// TODO possibly make preprocessed data have Optional fields
 fn save_and_preprocess_contract(
```

Comment:
This loop is starting to get a little yucky with a mix of IO and mutating references to the contracts. We'll want to have a way to preprocess a set of contracts without writing to disk in `noir_wasm`.

`noir_wasm` doesn't seem to connect to a backend atm however so not a blocker.